### PR TITLE
Multiple Y-Axes for Overlay Plots

### LINKED ISSUE #2321: [Plots] Allow multiple range axes in Overlay Plots
Allow a user to configure and display multiple range axes (Y axes currently) ticks for fields with different range values. All currently available controls for the range axis should be made available to multiple ones, including auto-scale | manual min-max. Issues around comparative units and scale need to be considered: are there constraints imposed to ensure that the range axis plots at a meaningful scale, or is it up to the user to ensure they choose correctly when configuring the plot?

From Rick Elphic, 2015:
> Plots of two quantities having different y-scale values can be very handy. In other words, a double-Y plot. Example: if I want to monitor two quantities of very different value ranges that may have a relationship, then having them on the same plot panel makes good use of screen real estate. 

Relates to [VISTA 241](https://github.jpl.nasa.gov/MissionControl/vista/issues/241) 
@mudinthewater: any additional thoughts?

### PR DESCRIPTION
**Pull request by ozyx**
Closes #2321 

### Describe your changes:
For overlay plots:
1. Extend elements pool to include multiple y axes (3 for now) - allows moving series between y axes

Model changes:
1. Update the plot series to include a yAxisId so it can map it to a yAxis
2. Update the y axis model to have an id and to update it's stats/display range based on which series is assigned to it.
3. PlotConfigurationModel changes to load any saved configuration for the plot

Plot and Chart changes:
1. Update Y Axis and Ticks to display more than one y Axis based on whether they are associated with a series.
2. Update the Plot to maintain yScales for each y axes and associated marquees and zoom/pan (TODO: could use optimization)
3. Update the Chart to maintain the offsets for each y axis and draw on the the same canvas. Only draw series for y axis that are visible.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/plot/MctPlot.vue [hunk 1]
```diff
@@ -269,13 +278,14 @@ export default {
         }
     },
     data() {
+        this.yAxisIdVisibility = {};
```
Comment (akhenry) [inline src/plugins/plot/MctPlot.vue:281]: please move to return block.

File: src/plugins/plot/axis/YAxis.vue [hunk 2]
```diff
@@ -63,67 +73,163 @@
...
+        },
+        position: {
+            type: String,
+            default() {
+                return 'left';
             }
         }
     },
     data() {
+        this.seriesModels = [];
```
Comment (akhenry) [inline src/plugins/plot/axis/YAxis.vue:118]: Please move to return block

File: src/plugins/plot/MctPlot.vue [hunk 3]
```diff
@@ -1181,9 +1287,16 @@ export default {
...
         endAnnotationMarquee(event) {
             const minX = Math.min(this.marquee.start.x, this.marquee.end.x);
-            const minY = Math.min(this.marquee.start.y, this.marquee.end.y);
+            const startMinY = this.marquee.start.y.reduce((previousY, currentY) => {
+                return Math.min(previousY, currentY);
+            }, this.marquee.start.y[0]);
+            const endMinY = this.marquee.end.y.reduce((previousY, currentY) => {
+                return Math.min(previousY, currentY);
+            }, this.marquee.end.y[0]);
+            console.log(startMinY, endMinY);
```
Comment (akhenry) [inline src/plugins/plot/MctPlot.vue:1296]: Let's remove this debug output.

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/plugins/plot/MctPlot.vue (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:281]: Hmm. Doesn't that make it reactive? We don't need it to be.

File: src/plugins/plot/MctPlot.vue (hunk 3 above)
Comment (shefalijoshi) [reply src/plugins/plot/MctPlot.vue:1296]: Done

File: src/plugins/plot/axis/YAxis.vue (hunk 2 above)
Comment (shefalijoshi) [reply src/plugins/plot/axis/YAxis.vue:118]: Doesn't that make it reactive? We don't need it to be.

**Review by akhenry — APPROVED** [review]

**Pull request merged**